### PR TITLE
Travis CI: Test Python 3.9 release candidate 1

### PR DESCRIPTION
**Pull request by cclauss**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

Travis CI: Test Python 3.9 release candidate 1
https://www.python.org/downloads/release/python-390rc1/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -113,21 +123,18 @@ before_install:
           wget -qO - https://packages.couchbase.com/ubuntu/couchbase.key | sudo apt-key add -
           sudo apt-add-repository -y 'deb http://packages.couchbase.com/ubuntu bionic bionic/main'
           sudo apt-get update && sudo apt-get install -y libcouchbase-dev
+          install: pip --disable-pip-version-check install --upgrade-strategy eager -U tox | cat
```
Comment (thedrow) [inline .travis.yml:126]: Why did you move this to before_install?

**Review by thedrow — CHANGES_REQUESTED** [review]

**Comment by thedrow** [comment]
It seems like we have a genuine test failure to investigate.

**Comment by cclauss** [comment]
Is the pypy3 failure caused by something in this PR?

**Comment by cclauss** [comment]
@untitaker FYI

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: That is the order in which things execute when the job is run.

**Comment by thedrow** [comment]
There's no need to allow failures for PyPy.
I've rebased your commits and it should now run as expected.

**Comment by thedrow** [comment]
This failure is very strange.

```python
    def test_append_to_empty_chain(self):
        x = chain()
        x = x | self.add.s(1, 1)
        x = x | self.add.s(1)
        x.freeze()
        tasks, _ = x._frozen
        assert len(tasks) == 2

        assert x.apply().get() == 3
```
passes the test but
```python
    def test_append_to_empty_chain(self):
        x = chain()
        x |= self.add.s(1, 1)
        x |= self.add.s(1)
        x.freeze()
        tasks, _ = x._frozen
        assert len(tasks) == 2

        assert x.apply().get() == 3
```
does not.
```

**Comment by thedrow** [comment]
I figured out why.
The `Signature` class inherits from `dict` which in Python 3.9 has the merge operator `|` and `|=` is the in place version of that operator.
We override `__or__` but not `__ior__` since previously `__ior__` would use the default implementation.

**Comment by thedrow** [comment]
Thank you so much!
